drivers/dma/dma_stm32.c

```diff
@@ -287,6 +287,16 @@ static int dma_stm32_configure(struct device *dev, u32_t id,
 	stream->src_size	= config->source_data_size;
 	stream->dst_size	= config->dest_data_size;
 
+	if ((&config->head_block->source_address == NULL)
```

Comment:
What is the meaning of this check? Test if config->head_block is NULL? So, why not checking it in this way?
Moreover, if this is what you want to check, maybe is better to test it in the beginning of the routine.

Reply:
We control here that the buffer for receiving or sending data is not 0.
 In case it is 0, it means the transfer will read (or write) mem address 0x0000 0000, which is not actually an error in terms of DMA memory transfer. 
Yes, this can be moved above.

Reply:
Wait a moment.
I think that &config->head_block->source_address is always != from 0, isn't it?
THis is the address of a field in a structure. so != from 0.
I guess that check may be only:
if (config->head_block->source_address != 0)

What do you think?

Reply:
That was the purpose of the testcase: `spi_null_tx_buf` in tests/drivers/spi/spi_loopback
to assign set 0 for the transmit (origin) mem address. and the structure tx_bufs is valid.
Here is what I want to check.

Reply:
OK, so do you agree that if (config->head_block->source_address != 0) should be enough?

Reply:
yes, agreed